app/Commands/Sites/CreateCommand.php

```diff
@@ -70,58 +46,76 @@ protected function action(): int
             return self::INVALID;
         }
 
-        $server    = $this->selectServer('deploy to')->first();
-        $userInput = $this->getUserInput();
-        $site      = $this->spinupwp->createSite($server->id, array_merge($this->arguments(), $this->options(), $userInput));
+        $server       = $this->selectServer('deploy to')->first();
+        $this->domain = $this->option('domain') ?? $this->resolveAnswer(['prompt' => 'Domain Name'], $this->nonInteractive());
+        $userInput    = $this->promptForAnswers($this->nonInteractive());
+
+        $site = $this->spinupwp->createSite($server->id, array_merge($this->arguments(), $this->options(), ['domain' => $this->domain], $userInput));
 
         $this->successfulStep("{$site->domain} is {$site->status} (event_id = {$site->eventId()})");
 
         return self::SUCCESS;
     }
 
-    /**
-     * Use different options depending on site install type.
-     */
-    protected function getAvailableOptionIO(string $type): array
+    protected function getPrompts(): array
     {
-        switch ($type) {
+        $prompts = [
+            'https_enabled' => [
+                'default'               => 1,
+                'nonInteractiveDefault' => 0,
+                'type'                  => 'confirm',
+                'prompt'                => 'Enable HTTPS',
+            ],
+            'site_user' => [
+                'prompt'          => 'Site User',
+                'defaultCallback' => [$this, 'getDomainSlug'],
```

Comment:
Why a callback? Why not
```php
'default' => $this->getDomainSlug(),
```

Reply:
Yup, that'll work. This is just an artifact of my refactoring process.

Reply:
✅